src/simularium/VisData.ts

```diff
@@ -369,36 +376,44 @@ class VisData {
      *   Functions to check update
      * */
     public hasLocalCacheForTime(timeNs: number): boolean {
+        const firstFrameTime = this.frameDataCache[0].time;
+        const lastFrameTime = this.frameDataCache[
+            this.frameDataCache.length - 1
+        ].time;
+
+        // Edge cases
         if (
             this.frameDataCache.length > 0 &&
             timeNs === 0 &&
-            this.frameDataCache[0].time <= Number.EPSILON // allow for floating point errors
+            firstFrameTime <= Number.EPSILON // allow for floating point errors
```

Comment:
interesting: first frame time is assumed to be <= 0

Reply:
Oh true! But it should be ok since all other cases are handled by the final return statement `return notLessThanFirstFrameTime && notGreaterThanLastFrameTime;`. I'm not 100% sure these if statements are necessary, but I'll just keep them in for now?